### DISCUSSION TITLE
make the OCR operation accessible manually

### DISCUSSION THREAD

**Discussion by mortee** [R0Wi-DEV/workflow_ocr#198] [Ideas]
While having the ability to automatically OCR files in Flow, it would actually make sense to make this available manually, from the UI. It could e.g. be an item in the context menu of supported file types. 

There's cases where I don't want to set up a complete workflow, I just need some file to be OCR'ed. Also, when I _do_ want to set up a workflow, but I already have a bunch of files there before setting it up, it would be nice to be able to process those manually (yes, a whole selection in a single operation, too).

I guess all the settings could be taken from where the Flow version does. But with manual invocation, it might not make sense to keep those settings at the bottom of the Flow settings page.

**Comment by R0Wi** [comment]
In general that sounds like a nice idea to me. Depending on how many settings you want to be able to apply before running the manual OCR, this could be a bit of work though.

From an MVP perspective I'd develop this gradually like:

1. Making a context menu item "Process via OCR" available for a single file. This will process the file with some default settings which cannot be tweaked in a first draft.
2. Making the same menu item available when multiple files are selected. Again applying default settings.
3. If 1 and 2 are implemented, introduce the "Per Workflow Settings" UI to the user after he clicked on the menu item. Do the OCR process with the given settings
4. Add some loading screen UI / progress information

From a processing perspective I think that processing the files synchronously would be the best and easiest idea. This would lead to the fact that an user would need to wait until the OCR process has finished after he clicked on the menu item. 

What are your thoughts? Any coding help would also be appreciated :+1:

**Comment by mortee** [comment]
Yes, the generic approach seems fine to me. For what it's worth, I wouldn't even expect a separate UI, but if it's feasible, sure, why not?

As for the synchronous execution: I'd basically use the same job submission / progress indication method which is used by other file operations. That way the user can see something is happening on their behalf, and how it is going.

When it comes to coding help: I haven't developed anything for NC yet, but if you're willing to give some guidance, then I might be up for it.
